internal/controllers/addoninstance_watcher.go

```diff
@@ -0,0 +1,65 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/openshift/reference-addon/internal/addonsdk"
+)
+
+type AddonInstanceWatcher struct {
+	client.Client
+	*addonsdk.StatusReporter
+	Log logr.Logger
+}
+
+func (r *AddonInstanceWatcher) Reconcile(
+	ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	addonInstance := &addonsv1alpha1.AddonInstance{}
+	if err := r.Get(ctx, req.NamespacedName, addonInstance); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	r.Log.Info("reporting addon-instance spec change to status-reporter")
+	if err := r.StatusReporter.ReportAddonInstanceSpecChange(ctx, *addonInstance); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *AddonInstanceWatcher) SetupWithManager(mgr ctrl.Manager) error {
+	addonInstanceConfigurationChangePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
```

Comment:
This will render this reconciler inoperable, because it will filter the initial "has been recorded/created in cache" event, also cache re-syncs will be filtered.
The AddonOperator would have to send an explicit update spec after the Operator is started, or this reconciler would never trigger `.Reconcile()`

Reply:
Ohh nice catch, thanks :)

Reply:
btw, I implemented and pushed your suggestion but I'd still love to get these above points of mine to be clarified by you, Thanks :)